akka-actor-tests/src/test/java/akka/actor/JavaExtension.java

```diff
@@ -34,7 +34,11 @@ public TestExtension(ExtendedActorSystem i) {
   }
 
   static class OtherExtension implements Extension {
-    static final ExtensionKey<OtherExtension> key = new ExtensionKey<OtherExtension>(OtherExtension.class) {
+    static final ExtensionId<OtherExtension> key = new AbstractExtensionId<OtherExtension>() {
```

Comment:
this should also `implements ExtensionIdProvider` http://doc.akka.io/docs/akka/2.4/java/extending-akka.html